crates/ruff_python_semantic/src/context.rs

```diff
@@ -387,31 +386,20 @@ impl<'a> Context<'a> {
 
     /// Returns the current top most scope.
     pub fn scope(&self) -> &Scope<'a> {
-        &self.scopes[self.scope_stack.top().expect("No current scope found")]
-    }
-
-    /// Returns the id of the top-most scope
-    pub fn scope_id(&self) -> ScopeId {
```

Comment:
I removed this method since `scope_id` is still `pub`, and it's confusing (IMO) to have a getter for a public field.

It's hard to use this method everywhere (same with some of the utilities, like `scopes`) because of the borrow checker semantics. Often, we borrow (e.g.) `self.ctx.scope_id`, then mutate `self.ctx.bindings` or something similar. The borrow checker is fine with that. But if we use `self.ctx.scope_id()`, we can't mutate other parts of `self.ctx`, since that's considered an immutable borrow of `self.ctx`.

Reply:
Hmm, `scope_id` shouldn't borrow because it's copied. I do see value in migrating towards getters to avoid our very loose APIs and long getter chains to context-specific data structures.

Reply:
Oh, hmm, perhaps that does work here, I have to check. (It doesn't work in general for references, but you're right that I'd expect it to work with copies.)

I don't disagree, but if we add / retain this, I think we should try to remove `pub` from the `scope_id` attribute. Otherwise, it's unclear which access pattern to use (and editor completion is confusing too).